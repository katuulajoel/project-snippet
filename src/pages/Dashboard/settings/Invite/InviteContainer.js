/* -------------------------------------------------------------------------- */
/*                            External Dependencies                           */
/* -------------------------------------------------------------------------- */
import React from "react";
import PropTypes from "prop-types";
/* ------------------------- Component dependencies ------------------------- */
import SectionNav from "../../../../components/SectionNav";
import { isAdmin } from "../../../../utils/auth";

const InviteContainer = (props) => {
  return (
    <>
      <SectionNav
        style={{
          marginBottom: "0px",
        }}
        links={[
          ...(isAdmin
            ? [
                { route: `settings/invite`, name: "INVITE USERS" },
                {
                  route: `settings/invite/create`,
                  name: "CREATE CLIENT",
                },
                {
                  route: `settings/invite/pending`,
                  name: "View pending invites",
                },
              ]
            : [
                { route: `settings/invite`, name: "INVITE USERS" },
                {
                  route: `settings/invite/pending`,
                  name: "View pending invites",
                },
              ]),
        ]}
      />
      {props.children}
    </>
  );
};

InviteContainer.propTypes = {
  children: PropTypes.element,
};

export default InviteContainer;
