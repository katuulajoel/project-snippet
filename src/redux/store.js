import { applyMiddleware, createStore } from "redux";
import reduxThunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";

import monitorReducersEnhancer from "../configs/enhancers/monitorReducer";
import loggerMiddleware from "../configs/middleware/logger";
import rootReducer from "./reducers";

export function configureStore(preloadedState = {}) {
  const middlewares = [loggerMiddleware, reduxThunk];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const enhancers = [middlewareEnhancer, monitorReducersEnhancer];
  const composedEnhancers = composeWithDevTools(...enhancers);

  const store = createStore(rootReducer, preloadedState, composedEnhancers);

  if (process.env.NODE_ENV !== "production" && module.hot) {
    module.hot.accept("./reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
}

const store = configureStore();
export default store;
