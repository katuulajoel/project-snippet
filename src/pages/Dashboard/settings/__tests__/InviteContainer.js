import React from "react";
import { Provider } from "react-redux";
import { cleanup, render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import InviteContainer from "../Invite/InviteContainer";
import store from "../../../../redux/store";

afterEach(cleanup);

describe("TabBar comp test", () => {
  it("TabBar component snapshot", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <InviteContainer />
        </Router>
      </Provider>
    );

    expect(asFragment(<InviteContainer />)).toMatchSnapshot();
  });
});
