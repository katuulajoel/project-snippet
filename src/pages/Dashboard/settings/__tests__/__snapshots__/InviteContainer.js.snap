// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBar comp test TabBar component snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="sc-bdnxRM gOFDIS"
      style="margin-bottom: 0px;"
    >
      <div
        class="filter-links"
      >
        <a
          href="/settings/invite"
        >
          INVITE USERS
        </a>
        <a
          href="/settings/invite/create"
        >
          CREATE CLIENT
        </a>
        <a
          href="/settings/invite/pending"
        >
          View pending invites
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
